lib/data_services_api/sapint_converter.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+module DataServicesApi
+  class SapiNTConverter
+    def initialize(dsapi_query)
+      @dsapi_query = JSON.parse(dsapi_query)
+    end
+
+    # Converts a DSAPI query to SAPINT query
+    def to_sapint_query
+      result = @dsapi_query.map do |key, value|
+        sapint_query(key, value)
+      end.compact.join('&')
+      sanitize_result(result)
+    end
+
+    private
+
+    def sapint_query(key, value)
+      case key
+      when '@sort'
+        sort(value)
+      when '@count'
+        count(value)
+      when '@limit'
+        limit(value)
+      when '@offset'
+        offset(value)
+      when '@and'
+        andd(value)
+      end
+    end
+
+    def sort(values)
+      values.map do |value|
+        sort_prop = value.key?('@up') ? "+#{value['@up']}" : "-#{value['@down']}"
+        "_sort=#{sort_prop}"
+      end
+    end
+
+    def count(value)
+      '_count=@id' if value == true
+    end
+
+    def limit(value)
+      "_limit=#{value}"
+    end
+
+    def offset(value)
+      "_offset=#{value}"
+    end
+
+    def andd(values)
```

Comment:
The logic here is a bit hard to follow with the use of `value`, `values` and `[]` indexing. Can you please refactor to make the logic a bit clearer?